File: TP1/C/Palindromo/Palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

int main() {
    char palavra[100];

    while (true) {
        fgets(palavra, sizeof(palavra), stdin);
        palavra[strcspn(palavra, "\n")] = '\0';

        if (strcmp(palavra, "FIM") == 0) {
            break;
        }

        int tam = strlen(palavra);
        bool pali = true;

        for (int i = 0; i < tam / 2; i++) {
            if (palavra[i] != palavra[tam - i - 1]) {
                pali = false;
                break;
            }
        }

        if (pali) {
            printf("SIM\n");
        } else {
            printf("NAO\n");
        }
    }

    return 0;
}

File: TP1/Java/Palindromo/Palindromo.java
import java.util.Scanner;

public class palindromo {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in); 
        
        do {
            String p = entrada.nextLine();
            
            if (p.equals("FIM")) {  
                break;  
            }

            int tam = p.length(); 
            boolean pali = true;

            for (int i = 0; i < tam / 2; i++) {  
                if (p.charAt(i) != p.charAt(tam - i - 1)) {  //funciona como um array[i] e array[tam-i-1]
                    pali = false;
                    break;  
                }
            }

            if (pali) {
                System.out.println("SIM");
            } else {
                System.out.println("NAO");
            }

        } while (true);  

        entrada.close(); 
    }
}
